### ISSUE TITLE
Topic model by_most_recent_post seems wacky

### ISSUE THREAD

**Issue by skipperguy12** [NJayDevelopment/mongoid_forums#12]
http://mongoid-forums.herokuapp.com/forums/

The "what's new" section uses the following snippet of code to fetch the most recent topics. It overrides mongoid_forums/forums#index:
```
      @recent = Topic.by_most_recent_post.page(params[:page]).per(MongoidForums.per_page)
```

I suspect this is because last_post isn't getting set.

**Comment by AustinLMayes** [comment]
You need to change the 'before_save' to a 'before_create' in the post model. The date displayed is the creation date not the update date.

**Comment by skipperguy12** [comment]
@AustinLMayes oops, thanks! Will try to fix this as soon as I can.

**Comment by skipperguy12** [comment]
Oh I had fixed this in a commit earlier but didn't push, my bad...

**Comment by skipperguy12** [comment]
closed in 201a3beaf12d2db8d3aa7a20a8aad6bace7f4332

**Comment by skipperguy12** [comment]
Pushed fixed version to heroku.

Although no one probably really had this issue due to how few people are actually using this gem right now, if you want a quick fix to get your topics updated (ones where the last post was not before this patch) then please run the following script in your rails console:
```
MongoidForums::Topic.each do |t|
  if t[:last_post_at] == nil
    update_attribute(:last_post_at, t.posts.last.created_at)
  end
end
```
